juju/juju
# Remove machine hasvote attribute

## Description of change

Previously a new controller node entity was added which has the "has vote" and "wants vote" attributes. This PR removes the now defunct "has vote" attribute from the machine entity, which was just mirroring the controller node value.

The second commit removes obsolete and broken code which was added 4 years ago to allow mongo to be upgraded prior to juju 2.x. The code was untested and the facade api calls weren't wired up correctly in once case due to a typo.

## QA steps

```
bootstrap lxd
enable-ha
wait for ha to be set up, watching show-controller and mongo rs.config()
remove machine 0
check show-controller and mongo rs.config()
add a new machine
enable-ha --to X
wait for ha to be set up, watching show-controller and mongo rs.config()
enable-ha -n 5
wait for ha to be set up, watching show-controller and mongo rs.config()
```
